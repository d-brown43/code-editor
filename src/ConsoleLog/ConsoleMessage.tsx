import * as React from "react";
import styles from "./ConsoleLog.module.scss";


type ConsoleMessage = {
    message: any;
}

const ConsoleMessage = ({message}: ConsoleMessage) => {
    const formattedMessage = React.useMemo(() => (
        <pre className={styles.formattedMessage}>
            {JSON.stringify(message, null, 2)}
        </pre>
    ), [message]);
    if (typeof message === 'string') {
        return <span>{message}</span>;
    }
    return formattedMessage;
};

export default ConsoleMessage;