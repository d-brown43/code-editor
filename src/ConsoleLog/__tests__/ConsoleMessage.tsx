import ConsoleMessage from "../ConsoleMessage";
import {shallow} from "enzyme";
import * as React from "react";

it('matches snapshot', () => {
    const consoleMessage = shallow(
        <ConsoleMessage message={'Hello World'} />
    );
    expect(consoleMessage).toMatchSnapshot();
});

it('matches snapshot', () => {
    const consoleMessage = shallow(
        <ConsoleMessage message={{a: 1, b: 2}} />
    );
    expect(consoleMessage).toMatchSnapshot();
});

it('matches snapshot', () => {
    const consoleMessage = shallow(
        <ConsoleMessage message={[1, 2, 3]} />
    );
    expect(consoleMessage).toMatchSnapshot();
});

it('matches snapshot', () => {
    const consoleMessage = shallow(
        <ConsoleMessage message={undefined} />
    );
    expect(consoleMessage).toMatchSnapshot();
});

it('matches snapshot', () => {
    const consoleMessage = shallow(
        <ConsoleMessage message={null} />
    );
    expect(consoleMessage).toMatchSnapshot();
});
