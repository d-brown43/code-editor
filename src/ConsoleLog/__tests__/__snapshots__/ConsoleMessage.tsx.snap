// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span>
  Hello World
</span>
`;

exports[`matches snapshot 2`] = `
<pre
  className="formattedMessage"
>
  {
  "a": 1,
  "b": 2
}
</pre>
`;

exports[`matches snapshot 3`] = `
<pre
  className="formattedMessage"
>
  [
  1,
  2,
  3
]
</pre>
`;

exports[`matches snapshot 4`] = `
<pre
  className="formattedMessage"
/>
`;

exports[`matches snapshot 5`] = `
<pre
  className="formattedMessage"
>
  null
</pre>
`;
